src/containers/BooksForm.js

```diff
@@ -1,27 +1,69 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
 import { categories } from '../constant';
+import { createBook } from '../actions/index';
 
-const BooksForm = () => (
-  <form>
-    <div className="form-group">
-      <input type="text" className="form-control" id="title" placeholder="Title" />
-    </div>
-    <div className="form-group">
-      <select className="form-control" id="category">
-        {
+const mapDispatchToProps = dispatch => ({
+  createBook: book => dispatch(createBook(book)),
+});
+
+const BooksForm = ({
+  createBook,
+}) => {
+  const intialstate = {
+    title: '',
+    category: categories[0],
+  };
+
+  const [state, setstate] = useState(intialstate);
+
+  const handleInputChange = event => {
+    setstate({
+      ...state,
+      title: event.target.value,
+    });
+  };
+
+  const handleSelectChange = event => {
+    setstate({
+      ...state,
+      category: event.target.value,
+    });
+  };
+
```

Comment:
[Required] We can remote one of these functions as they perform the same task. You can pass the name and value argument to a function and update the state according to the name.
For example,
const handleChange = (name, value) => {
name: value 
}
something like this.